doc/code_snippets/test/msgpack/msgpack_object_iterator_map_test.lua

```diff
@@ -0,0 +1,20 @@
+local msgpack = require('msgpack')
+
+local mp_map = msgpack.object({ foo = 'a',
+                                bar = { 'b1', 'b2', 'b3' } })
+local mp_map_iterator = mp_map:iterator()
+
+local size = mp_map_iterator:decode_map_header() -- returns 2
+local first = mp_map_iterator:decode()           -- returns 'foo'
+local second = mp_map_iterator:decode()          -- returns 'a'
+mp_map_iterator:skip()                           -- returns none, skips 'bar'
+local fourth = mp_map_iterator:decode()          -- returns ['b1', 'b2', 'b3']
```

Comment:
Strictly speaking, map elements will not necessarily be returned in this order. A Lua map is unordered so the order in which it's encoded in MsgPack depends on Lua internals. It may be better to rework the example to put decoded elements back into a map/set.